presto-main/src/main/java/io/prestosql/server/security/KerberosAuthenticator.java

```diff
@@ -94,7 +98,7 @@ public KerberosAuthenticator(KerberosConfig config)
             loginContext.login();
 
             serverCredential = doAs(loginContext.getSubject(), () -> gssManager.createCredential(
-                    gssManager.createName(config.getServiceName() + "@" + hostname, GSSName.NT_HOSTBASED_SERVICE),
+                    getGSSName(config, hostname),
```

Comment:
We should grab the variables out of the config class outside of the lambda, so that we are not holding onto a (potentially) mutable config class inside of the lambda. (the existing code is incorrect)